docker/android.cmake

```diff
@@ -0,0 +1,33 @@
+# toolchain file for android targets, see #1110
+
+set(CMAKE_SYSTEM_NAME Android)
+set(CMAKE_SYSTEM_PROCESSOR "$ENV{CROSS_ANDROID_ARCH}")
+set(CMAKE_ANDROID_STANDALONE_TOOLCHAIN /android-ndk)
+set(CMAKE_ANDROID_API "$ENV{CROSS_ANDROID_SDK}")
+
+# not all of these are standard, however, they're common enough
+# that it's good practice to define them.
+set(prefix "$ENV{CROSS_ANDROID_TARGET}")
+set(CMAKE_C_COMPILER "${prefix}-gcc")
```

Comment:
Only `CMAKE_AR`, `CMAKE_RANLIB`, `CMAKE_C_COMPILER`, `CMAKE_ASM_COMPILER`, and `CMAKE_CXX_COMPILER` are standard, however, other tools will often use values provided by those variables even though they're not standard. Many toolchain files will provide them.

Reply:
It turns out setting these manually for the `CMAKE_ASM_COMPILER`, etc., fails badly with downstream configurations. See https://github.com/cross-rs/cross/issues/1110#issuecomment-1303781124.